### PR TITLE
tests: spread test for snap refresh/switch channel and risk switching

### PR DESCRIPTION
**Pull request by stolowski**
Based on #7357. Add a spread test for snap switch/refresh and new channel/risk switching semantics introduced with already landed #7199.

Uses lxd at the moment, I've requested a track for test-snapd-tools snap and will update the test to use it when available.

### CODE REVIEW

File: tests/main/snap-switch/task.yaml [hunk 1]
```diff
@@ -4,5 +4,8 @@ execute: |
     snap install test-snapd-tools
     snap info test-snapd-tools|MATCH "tracking: +stable"
 
-    snap switch --edge test-snapd-tools
+    snap switch --edge test-snapd-tools > stdout.txt
     snap info test-snapd-tools|MATCH "tracking: +edge"
+
+    echo "Ensure we don't print incorrect and confusing information"
+    not grep "is closed; temporarily forwarding to stable." < stdout.txt
```
Comment (anonymouse64) [inline tests/main/snap-switch/task.yaml]: can we use `MATCH` here or is there a reason to use `grep`?

File: tests/main/snap-track-risk/task.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/edge"
+    snap refresh "$SNAP" --channel=latest/stable
+    snap info "$SNAP" | MATCH "tracking: +latest/stable"
+
+    snap switch --channel="$TRACK/edge" "$SNAP"
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/edge"
+    snap switch --channel=stable "$SNAP"
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/stable"
+    snap switch --channel=latest/stable "$SNAP"
+    snap info "$SNAP" | MATCH "tracking: +latest/stable"
```
Comment (anonymouse64) [inline tests/main/snap-track-risk/task.yaml]: minor nitpick, missing newline at EOF

**Comment by anonymouse64** [comment]
The unit tests failed oddly here:

```
=== RUN   TestConnect
--- FAIL: TestConnect (0.00s)
	conn_test.go:10: unable to subscribe to netlink uevent, err: address already in use
=== RUN   TestRules
--- PASS: TestRules (0.00s)
=== RUN   TestParseUEvent
--- PASS: TestParseUEvent (0.00s)
=== RUN   TestParseUdevEvent
--- PASS: TestParseUdevEvent (0.00s)
FAIL
coverage: 55.8% of statements
FAIL	github.com/snapcore/snapd/osutil/udev/netlink	0.003s
```

but I don't think that's related so I've gone ahead and restarted the test

File: tests/main/snap-track-risk/task.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    snap refresh "$SNAP" --channel="$TRACK/edge"
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/edge"
+    snap refresh "$SNAP" --stable
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/stable"
+    snap refresh "$SNAP" --channel=edge
+    snap info "$SNAP" | MATCH "tracking: +$TRACK/edge"
+    snap refresh "$SNAP" --channel=latest/stable
+    snap info "$SNAP" | MATCH "tracking: +latest/stable"
+
+    snap switch --channel="$TRACK/edge" "$SNAP"
```
Comment (anonymouse64) [inline tests/main/snap-track-risk/task.yaml]: I think we should also add a check for `snap switch --edge "$SNAP"`, etc. to match what we have for the refresh case, what do you think?

**Comment by zyga** [comment]
The spread test needs some love:
```
2019-08-29 22:17:57 Error executing google:ubuntu-19.04-64:tests/main/snap-info : 
-----
+ echo 'With no arguments, errors out'
With no arguments, errors out
+ not snap info
error: the required argument `<snap> (at least 1 argument)` was not provided
+ echo 'With one non-snap argument, errors out'
With one non-snap argument, errors out
+ not snap info /etc/passwd
error: no snap found for "/etc/passwd"
+ snap info --unicode=always basic_1.0_all.snap /home/gopath/src/github.com/snapcore/snapd/tests/lib/snaps/basic-desktop test-snapd-tools test-snapd-devmode core /etc/passwd test-snapd-python-webserver
+ PYTHONIOENCODING=utf8
+ python3 check.py
in test-snapd-tools.channels: extra keys: {'2.0/beta', '2.0/candidate', '2.0/stable', '2.0/edge'}
```

**Comment by stolowski** [comment]
Closing as it is not very useful due to revert of the base changes.

**Pull request closed without merging**